adminFunctions.php

```diff
@@ -0,0 +1,45 @@
+<?php
+require_once('cmsController.php');
+$db = callDatabase();
+
+/**
+ *The function retrieves the password from the database based off the userName, de-hashes DB password and compares to input password
+ *
+ * @param $enteredPassword string value
+ * @param $enteredUName string value
+ * @param $db PDO to draw from
+ *
+ *@return boolean result to confirm passwords match
+ */
+
+function pullAndComparePasswords(string $enteredPassword, string $enteredUName, PDO $db):bool {
+    $query = $db->prepare("SELECT `password` FROM `users` WHERE `user` = :uName;");
+    $query->bindParam(':uName', $enteredUName);
+    $query->execute();
+    $passwordDB = $query->fetch();
+    $boolX =  password_verify($enteredPassword, $passwordDB['password']);
+    return $boolX;
+}
+
+function stripPassword(string $password):string {
+    $string1 = trim($password);
+    $string2 = htmlspecialchars($string1);
+    $newPassword = stripslashes($string2);
```

Comment:
just return this line instead of creating a new variable. save memory, save the children, save the planet

Reply:
resolved